### ISSUE TITLE
Create CardTopicSelection component

### ISSUE THREAD

**Issue by andrewtavis** [activist-org/activist#276]
### Terms

- [X] I have searched [open and closed feature requests](https://github.com/activist-org/activist/issues?q=is%3Aissue+label%3Afeature)
- [X] I agree to follow activist's [Code of Conduct](https://github.com/activist-org/activist/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

This issue would work on creating the topic selection component where a user could indicate topics that they, an organization or an event want to focus on (climate change, etc). The current version of this component can be seen in the [designs on Figma](https://www.figma.com/file/I9McFfaLu1RiiWp5IP3YjE/activist_public_designs?type=design&node-id=2475%3A10600&mode=design&t=hJRakNbmbK83mZ9P-1).

> [!NOTE]\
> This component needs a redesign! The first part of this issue is discussing how best for this component to be laid out for easy selection from what will eventually be ~20 topics.

EDIT: the most recent redesign of this component is below:

![Input Field](https://github.com/activist-org/activist/assets/24387426/c72d33c0-2f0f-4994-a737-a3239c895408)

Feedback on how best to design this component would be welcome, and after this the component can be created in the `components/card` directory.

### Contribution

Will be working on this, and would appreciate support from the community and people new to activist who want to share some ideas!

**Comment by TeddyGavi** [comment]
@andrewtavis I had some thoughts on this! :brain: 

Is the image associated with each topic needed?

Could we use a autocomplete select, where each selected topic is removed from the autocomplete list and added to a "topic list" displayed? Im thinking similar to what dev.to does when adding tags. (allows for image, and extensibility)

[Screencast from 2023-08-18 10:30:36 AM.webm](https://github.com/activist-org/activist/assets/102203363/df6765c0-9b99-4a4b-88e1-5fb4f79ea2c9)

Alternatively, a simple list of Radio buttons comes to mind. 

I would be interested in the accessibility of the first approach. FYI dev.to does a great job of this when I manually tested keyboard navigation.

**Comment by andrewtavis** [comment]
Hey @TeddyGavi! Wow sorry, received this while I was at Chaos Communication Camp and actually sat down to write replies to it a couple times 🤦‍♂️ 

> Is the image associated with each topic needed?

I'd say even just a little icon would be nice. These things are important for branding purposes, but we don't necessarily HAVE to have them, and certainly not at the start.

I do generally like this idea :) So we're talking an area where the the ones that have been selected are that's also a text box, and then there'd be a dropdown below it. Big thing is that we don't have too many options, so maybe having them all appear below in a grid would be fine? Let me design something for this and maybe we can discuss in the sync (if you'll be there)?

**Comment by andrewtavis** [comment]
@TeddyGavi, How does the new version of the component that I added above look to you? As we currently only have 21 topics, and the list is not expected to grow too much, I think that having all of them as selectable options beneath would be ok. Also makes them more recognizable with the topic markers that appear elsewhere on the platform :) Functionality would be similar to the video you shared above: a person could both select a topic from the dropdown and also enter in the name as text with maybe the next that's alphabetical in the list of the unselected ones then being able to be autocompleted.

Let me know what you think!

**Comment by TeddyGavi** [comment]
@andrewtavis this is great! I really like this design, modern and simple. The topic suggestion text does lead me to believe that the topics will somehow grow and change over time. If there is only going to be ~21, How does the suggestion work in that case? 

The new design is great! Amazing work on that. It really flows well with the rest of the components.

**Comment by TeddyGavi** [comment]
Hello @andrewtavis My apologies for being absent from this. Please assign this to me I would like to make a go at it!

:rocket:

**Comment by andrewtavis** [comment]
No stress on a bit of absence, @TeddyGavi! Assigning now, and let me know if you need any support!

**Comment by andrewtavis** [comment]
Hey @TeddyGavi! Hope you're well :) Also really hope you can make Code Night 🌙 tomorrow! The team's been talking about how you've been missed 😊

How are you doing on this issue? Let me know if there's anything I can do to help! :)

**Comment by TeddyGavi** [comment]
Hello! I am just writing a reply here when you messaged! :) . I have not been active on this issue, :( Please feel free to assign it to someone else. At this point, I cannot commit time to it :)

I am hoping to make time to be there, not guaranteed yet. I have been missing the welcomeness and active collaboration here! Talk soon.

**Comment by andrewtavis** [comment]
I’ll unassigned you, @TeddyGavi, but thanks for helping with the designs! There are a bunch of new issues that are not as involved if you wanted to pick one up 😊

**Comment by ZuzkaL** [comment]
Hi, I would love to work on this issue. Can you assign me?

**Comment by andrewtavis** [comment]
Hey @ZuzkaL 👋 Thanks for your interest in activist and helping out! I think that @momanyisamuel has a local version of this that he's planning on sending. Let's wait on that for a sec and maybe you can do some cleanup changes to it if needed?

**Comment by ZuzkaL** [comment]
Hey @andrewtavis ,
Thank you for the update! I'm glad I can help with the activism project and contribute. 
Additionally, I would like to make any other component that is needed. Feel free to assign me any other issue. I'd be happy to take on an issue where I can create a frontend component.

**Comment by andrewtavis** [comment]
Thanks for the interest and the dedication, @ZuzkaL! We'll be making more issues soon, and I'll keep you in mind and let you know when they're up. Will also update the Development channel that we have on [Matrix](https://matrix.to/#/%23activist_community%3Amatrix.org) if you wanted to join that :)

**Comment by andrewtavis** [comment]
Hey @bmartins-42 👋 c06ab9a extracts the code on [activist.org/organizations/create](https://activist.org/en/organizations/create) to a component [CardTopicSelection](https://github.com/activist-org/activist/blob/main/frontend/components/card/CardTopicSelection.vue).  Below is the updated design for it as well. Would doubtless be good to use some sort of Headluss UI for the search like maybe a [combobox](https://headlessui.com/vue/combobox) that takes in the available topics so the user can also type and select one. We'd also have all the topics beneath so they can select them, with selected ones moving to the front of the selected order (orange ones first) :) Happy to split this into multiple issues if that would help! Let me know if you have some ideas on this 😊

![Topic Selection](https://github.com/activist-org/activist/assets/24387426/ff86a419-fd42-4818-80be-d9d3bb42e2c9)

**Comment by andrewtavis** [comment]
Sounds great, @bmartins-42! Looking forward to seeing the results here 😊

**Comment by bmartins-42** [comment]
Ok, so far here are the question I got from working on this components

- Where does the topic list come from ? Backend  ? Configuration file, Constant ?
- Does the topi icon (currently globe) are always the same ?
- Why does some topic have another color ?
- Does the user allowed to create values ?
- What color do I use in the combobox for selected value ? And focused value ?
- Should I let the compobox list be over the topics tag list ?
- Should I set a maximum amount of element to be visible in the combobox and let the rest be scrolled ?

If you think the component is heading too far from the maquette, I could try to work without headlessUI combobox.

<br />

Here are some screenshots :

![image](https://github.com/activist-org/activist/assets/56431228/4cbd417e-eabe-48ac-9edf-44e8f4ea2c9d)

![image](https://github.com/activist-org/activist/assets/56431228/2988ed34-eea8-4899-bd3f-36a6adbb8673)

![close-topic-selection-with-no-topics](https://github.com/activist-org/activist/assets/56431228/53534331-6253-41c2-b2b6-54fc0999df47)

![open-topic-selection-with-2-topics](https://github.com/activist-org/activist/assets/56431228/9d19e3b6-cda4-4c3c-b577-028af00b3b2e)

**Comment by andrewtavis** [comment]
Hey @bmartins-42! Sorry for the wait on the answer. Had a bit of a crunch with a task at work :)

Answers to your questions:

> - Where does the topic list come from ? Backend ? Configuration file, Constant ?

Would come from the backend. The schema for it can be found [here](https://www.figma.com/file/I9McFfaLu1RiiWp5IP3YjE/activist_public_designs?type=design&node-id=5377%3A96951&mode=design&t=4kgbNlejVn8mC966-1), but for now you can just make a constant that we can replace :)

> - Does the topic icon (currently globe) are always the same ?

No. The general idea would be that we'd have a different icon for each of them. We've just been using the globe as a placeholder. I'm alright when it comes to designs, but the icons are pretty difficult to work out and there's still about half that need them. Let's use the globe for now and I'll make sure that we get them finished before beta 😊

> - Why does some topic have another color ?

Those that have a different color are those that are selected. The general idea is:
  
  - The user comes and they're all unselected and grey
  - They can type into the field to quickly select by typing and pressing return
  - They can also click on them individually
  - Once one is selected, it then moves to the front of the list and changes color (see below)

> - Does the user allowed to create custom topics ?

Topics no. As seen in the [schema](https://www.figma.com/file/I9McFfaLu1RiiWp5IP3YjE/activist_public_designs?type=design&node-id=5377%3A96951&mode=design&t=4kgbNlejVn8mC966-1), there is a `deprecation_date` such that we can change them, but generally the idea is that topics are set so there's some consistency and `"Tags"` can be used to be more explicit. I've yet to make the schema for tags, but just added it to the to do list :)

> - What color do I use in the combobox for selected value ? And focused value ?

For the selected value we'd be using `light-cta-orange dark:dark-cta-orange` and for the focussed we could do `light-cta-orange-hover dark:dark-cta-orange-hover`. Don't worry too much about the colors at this time though. I'm happy to clean it up :)

> - Should I let the compobox list be over the topics tag list ?

We can check for usability on this. Looking at how GitHub does label selection for issues, which is a similar feature to this, they actually filter out those elements that don't match the text that the user is typing. So if we had topics of "Environmentalism", "Democracy", etc, and I start typing "env" then the other topics would disappear and I could quickly arrow navigate to the one I want. This might be a bit much for now though, so no stress!

> - Should I set a maximum amount of element to be visible in the combobox and let the rest be scrolled ?

I'd say this makes sense for mobile, but it'd be good to see them all on desktop.

> If you think the component is heading too far from the maquette, I could try to work without headlessUI combobox.

Let's see if we can make it work, or if a different Headless component works better for this by all means feel free to give it a shot!

Also, if it works feel free to use `MarkerTopic` for the individual selectable items. We might need to expand the props for that though 🤔

Let me know if further questions come up! Really happy to have the support here! 🚀

**Comment by bmartins-42** [comment]
Hey @andrewtavis 

I finalized the V2, I reworked the combobox display to match the intent of the initial component (I didn't understand at first :sweat_smile: )

As often the color used probably need some tweak
I think we should have more contrast with the hover color, I added outline to improve the visibility, but the color is not one of the theme, so it should be changed

**Comment by andrewtavis** [comment]
Thanks for this, @bmartins-42! Sounds like we're ready for a PR here :) :) 

And yes the hover colors are definitely not what they should be. We need hover colors for the CTA orange, normal button color and the menu selections, and those we do have need more contrast to show up better. Do you want to open an issue for this? I can as well 😊

**Comment by bmartins-42** [comment]
> And yes the hover colors are definitely not what they should be. We need hover colors for the CTA orange, normal button color and the menu selections, and those we do have need more contrast to show up better. Do you want to open an issue for this? I can as well 😊

Sorry, I will let you create the issue, not sure how to present the subject,

**Comment by andrewtavis** [comment]
No stress whatsoever, @bmartins-42 :) I'll write it up soon 😊

**Comment by andrewtavis** [comment]
f9f00bc does some minor code formatting and updates `global` to `_global` so that these i18n keys will be first in the alphabetical order :) Some minor comments for another issue, @bmartins-42:

- Might be nice if on tab we also get down to the tops below
- It would be great if we could also use the left and right arrows to navigate between the topics on desktop!

Let me know on whether you think the above points are possible, and if so we can make a new issue 😊 

Thanks again so much!
